### ISSUE TITLE
Training with custom datasets

### ISSUE THREAD

**Issue by ghost** [mbaradad/shaders21k#5]
@mbaradad, I have created a very large synthetic dataset of my own, which is visually very different from the dataset in this and the previous paper. I am curious if the code in this repository allows for the use of any custom data other than what was used in this paper and your previous paper for training. Could you give me a clue if this is the right place for training? https://github.com/mbaradad/shaders21k/tree/main/sup_contrast
Thanks in advance

**Comment by mbaradad** [comment]
For the large-scale experiment (if you have 4GPU's with > 12GB) and can train for 1 week, you'll get comparable results to the ones in Table-3 of the main paper by running:
https://github.com/mbaradad/learning_with_noise/blob/main/scripts/train_moco/train_main_and_linear_moco.sh
You can replace the argument data/large_scale/$EXP_NAME for the directory with your training data, which has to have a train/XXXX folder with XXXX being subfolders with images (same structure as Imagenet, but the subfolders XXXX won't be interpreted as classes)

To replicate the results with those on Table 3, your dataset should contain 1.3M images and using the script above.

**Comment by ghost** [comment]
Thank you. I don't have 4GPU's with > 12GB, I have one with 8GB but I Have plenty of time. Thanks, I'll see if it runs in this situations.

**Comment by mbaradad** [comment]
That should be good. Results won't be exactly the same as the ones in the paper (because we don't use BatchNorm normalization when using 4GPU's), but should be on par (+-0.5% difference). For reasons we didn't investigate, the models tended to perform better without BatchNorm than with if I recall correctly.
